codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/CacheFileInteractionMode.scala

```diff
@@ -0,0 +1,5 @@
+package org.virtuslab.ash
+
+sealed trait CacheFileInteractionMode
+case class DumpTypesIntoCacheFile() extends CacheFileInteractionMode
```

Comment:
Nit: just case objects?

Reply:
Or better, just get rid of them completely, as suggested in SerializerCheckCompilerPluginComponent.scala

Reply:
File removed - according to other comment (no need to have these types as interactWithTheCacheFile has been replaced with two separate methods)